src/QMCWaveFunctions/Jastrow/DiffTwoBodyJastrowOrbital.h

```diff
@@ -139,7 +145,18 @@ class DiffTwoBodyJastrowOrbital : public DiffWaveFunctionComponent
         lapLogPsi[i]  = new ValueVectorType(NumPtcls);
       }
       OffSet.resize(F.size());
-      int varoffset = myVars.Index[0];
+
+      // Find first active variable for the starting offset
+      int varoffset = -1;
+      for (int i = 0; i < myVars.size(); i++)
+      {
+        varoffset = myVars.Index[i];
+        if (varoffset != -1)
+        {
+          break;
+        }
+      }
+
```

Comment:
If we have 3 distinct sets of coefficients (uu/ud/dd) and only the middle one is optimization. The code is still potentially wrong? Let me know your answer or your guess.

Reply:
I modified the unit test for this case, and it still passed.

Reply:
The unit test is only limited to check indices. So it is guaranteed up-to what is being checked. Not necessary a real J2 optimization will work. right?

Reply:
Yes, that's correct - this only fixes setting OffSet variable.  Shiv's PR had another change that I need to look at.

Reply:
I see thanks.